sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/testing/BigqueryMatcher.java

```diff
@@ -63,32 +63,52 @@
 
   private final String projectId;
   private final String query;
+  private final boolean usingStandardSql;
```

Comment:
Any reason why we need this boolean? IIUC, bq can tell whether current query is legacy or standard based on query string.

Reply:
No, BigQuery cannot tell that (or at least the BigQuery client library we use to issue the query cannot do that). We have to tune the parameter to specify which type of query we are using.
Link to the underlying client library code:
https://github.com/googleapis/google-api-java-client-services/blob/ed262e9fa7d950fa4e2aca65c2b765d544179859/clients/google-api-services-bigquery/v2/1.27.0/com/google/api/services/bigquery/model/QueryRequest.java#L122-L130
(called by BigQueryClient by BigQueryMatcher by the ITs)

Reply:
Thanks for the pointer! Then maybe we should setup `usingLegacySql` instead of `usingStandardSql` to keep consistent with bq model. wdyt?

Reply:
I have considered this. But in our Beam `BigQueryIO` source we also have
https://github.com/apache/beam/blob/08d0146791e38be4641ff80ffb2539cdc81f5b6d/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java#L600

Since this is a public API visible to Beam users but the BQ client is not, I decided to be consistent with the former (and therefore I have to negate the boolean somewhere in the function call stack).